### ISSUE TITLE
`azure.ai.ml.entities._inputs_outputs._get_param_with_standard_annotation` doesn't support stringizized annotations enabled by `from __future__ import annotations`.

### ISSUE THREAD

**Issue by mchikyt3** [Azure/azure-sdk-for-python#35538]
- **Package Name**: azure-ai-ml
- **Package Version**: 1.15.0
- **Operating System**: Windows 10
- **Python Version**: 3.12.2

**Describe the bug**
I'm using `mldesigner.command_component` to convert a function into a command component that is use later in a pipeline. The inputs and outputs of the command components are determined by the type annotations of the parameters of the function. However, stringized annotations can't be parsed, which is enabled by `from __future__ import annotations`.

After debugging, the problem can be traced down to [#L334](https://github.com/Azure/azure-sdk-for-python/blob/19bac3b570fbd55754b3c8e6f873663f1b5cf6e0/sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/utils.py#L334), where `getattr(cls_or_func, "__annotations__", {})` can be replaced with a helper function `typing.get_type_hints()` to correctly evaluate the type back from its string form.

**To Reproduce**
1. Run the following script:
```python
from __future__ import annotations

from azure.ai.ml.entities._inputs_outputs import _get_param_with_standard_annotation


def test_annotation(param: int):
    return


_get_param_with_standard_annotation(test_annotation, is_func=True)

```
2. The exception trace:
```
Exception has occurred: UserErrorException       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
Unsupported annotation type 'int' for parameter 'param'.
  File "C:\Users\someone\Documents\VSCode\.venvazureml\Lib\site-packages\azure\ai\ml\entities\_inputs_outputs\utils.py", line 142, in _get_fields
    raise UserErrorException(msg)
  File "C:\Users\someone\Documents\VSCode\.venvazureml\Lib\site-packages\azure\ai\ml\entities\_inputs_outputs\utils.py", line 337, in _get_param_with_standard_annotation
    annotation_fields = _get_fields(annotations)
                        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\someone\Documents\VSCode\test_annotation.py", line 10, in <module>
    _get_param_with_standard_annotation(test_annotation, is_func=True)
  File "C:\Program Files\Python312\Lib\runpy.py", line 88, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python312\Lib\runpy.py", line 198, in _run_module_as_main (Current frame)
    return _run_code(code, main_globals, None,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
azure.ai.ml.exceptions.UserErrorException: Unsupported annotation type 'int' for parameter 'param'.
```

**Expected behavior**
The type should be correctly recognized. I have tried `typing.get_type_hints` and it works fine.

**Comment by diondrapeck** [comment]
Hi @mchikyt3 - thank you for bringing this to our attention. I see your PR to address it. Please let me know if you need help getting the tests green. Once it is ready for review, I can review it and help you get it merged.

Thank you for contributing.

**Comment by mchikyt3** [comment]
Hi, @diondrapeck. Thank you for your reply. Regarding the test for the PR, I've found that the original code doesn't pass (see commit 261e93ec75c0ce449d5e64dc7d7027b246ddb1ce). Please let me know if this is a known issue, or I've missed something. Thanks.

**Comment by diondrapeck** [comment]
Can you link to the specific test/error log where you're seeing that? I don't see it in the errors for your PR.

**Comment by mchikyt3** [comment]
I have reverted my changes, only leaving an empty comment to trigger testing. You may check the errors now #35599. Thanks.

**Comment by diondrapeck** [comment]
Yes, I've confirmed this is a failure going on in main, unrelated to your PR. Please add your changes back and I'll review them. Once the main issue is fixed, we can get your PR tested objectively and merged. Thank you!
